dev/AppNotifications/ShellLocalization.cpp

```diff
@@ -0,0 +1,272 @@
+
+    return wicBitmapSourceConverted;
+}
+
+void AddFrameToWICBitmap(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapEncoder> const& wicEncoder,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    BITMAP_VERSION bitmapVersion)
+{
+    winrt::com_ptr<IWICBitmapFrameEncode> wicFrameEncoder;
+    winrt::com_ptr<IPropertyBag2> wicEncoderOptions;
+
+    THROW_IF_FAILED(wicEncoder->CreateNewFrame(wicFrameEncoder.put(), wicEncoderOptions.put()));
+
+    GUID containerGuid;
+    THROW_IF_FAILED(wicEncoder->GetContainerFormat(&containerGuid));
+
+    if ((containerGuid == GUID_ContainerFormatBmp) && (bitmapVersion == BITMAP_VERSION::VERSION5))
+    {
+        // Write the encoder option to the property bag instance.
+        VARIANT varValue{};
+        varValue.vt = VT_BOOL;
+        varValue.boolVal = VARIANT_TRUE;
+
+        // Options to enable the v5 header support for 32bppBGRA.
+        PROPBAG2 v5HeaderOption{};
+        v5HeaderOption.pstrName = (LPOLESTR) L"EnableV5Header32bppBGRA";
+
+        THROW_IF_FAILED(wicEncoderOptions->Write(1, &v5HeaderOption, &varValue));
+    }
+
+    THROW_IF_FAILED(wicFrameEncoder->Initialize(wicEncoderOptions.get()));
+
+    UINT width{};
+    UINT height{};
+    THROW_IF_FAILED(wicBitmapSource->GetSize(&width, &height));
+
+    THROW_IF_FAILED(wicFrameEncoder->SetSize(width, height));
+
+    winrt::com_ptr<IWICBitmapSource> wicbitmapSourceConverted
+        { ConvertWICBitmapPixelFormat(wicImagingFactory, wicBitmapSource, GUID_WICPixelFormat32bppBGRA, WICBitmapDitherTypeNone) };
+
+    WICRect rect{ 0 /* x-coordinate */, 0 /* y-coordinate */, static_cast<INT>(width), static_cast<INT>(height) };
+
+    // Write the image data and commit
+    THROW_IF_FAILED(wicFrameEncoder->WriteSource(wicbitmapSourceConverted.get(), &rect));
+    THROW_IF_FAILED(wicFrameEncoder->Commit());
+}
+
+winrt::com_ptr<IStream> GetStreamOfWICBitmapSource(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    REFGUID guidContainerFormat,
+    BITMAP_VERSION bitmapVersion)
+{
+    winrt::com_ptr<IStream> spImageStream;
+    THROW_IF_FAILED(CreateStreamOnHGlobal(nullptr /* handle */, true /* delete on release */, spImageStream.put()));
+
+    // Create encoder and initialize it
+    winrt::com_ptr<IWICBitmapEncoder> wicEncoder;
+    THROW_IF_FAILED(wicImagingFactory->CreateEncoder(guidContainerFormat, nullptr, wicEncoder.put()));
+
+    THROW_IF_FAILED(wicEncoder->Initialize(spImageStream.get(), WICBitmapEncoderCacheOption::WICBitmapEncoderNoCache));
+
+    // Add a single frame to the encoder with the Bitmap
+    AddFrameToWICBitmap(wicImagingFactory, wicEncoder, wicBitmapSource, bitmapVersion);
+
+    THROW_IF_FAILED(wicEncoder->Commit());
+
+    // Seek the stream to the beginning and transfer
+    static const LARGE_INTEGER lnBeginning = {};
```

Comment:
Why static?

Reply:
@sharath2727 can you help with this?

Reply:
We don't need the static. We are only dealing with one frame to convert to png. Got rid of it. Thanks.